### PR TITLE
Sugar for recipe2plan: making use of namespace to shorten particle class path

### PR DESCRIPTION
**Pull request by alxrsngrtn**
After this PR, if a particle path starts with `.`, we will prepend the namespace path. This will help developers write less verbose manifests.

### CODE REVIEW

File: src/tools/tests/recipe2plan-test.ts [hunk 1]
```diff
@@ -34,9 +37,93 @@ $ tools/update-goldens \n\n`
...
     const decoded = ManifestProto.decode(encoded);
     assert.lengthOf(decoded['recipes'], 1);
     assert.lengthOf(decoded['particleSpecs'], 1);
   }));
+  it('allows namespace to prefix particle path', Flags.withDefaultReferenceMode(async () => {
+    const manifest = await Runtime.parse(`
+    meta
+      namespace: arcs.core.data.testdata
+      
+    particle Writer in '.Writer'
```
Comment (bgogul) [inline src/tools/tests/recipe2plan-test.ts:51]: would it make sense to use `Writer` instead of `.Writer`?

Comment (alxrsngrtn) [reply src/tools/tests/recipe2plan-test.ts:51]: I thought about that. How would we differentiate between classes in the current directory with classes at the end of the path? 

The dot prefix is a simple way to disambiguate this. It also follows an Android Manifest convention, see [these docs](https://developer.android.com/guide/topics/manifest/manifest-intro#components).

Comment (alxrsngrtn) [reply src/tools/tests/recipe2plan-test.ts:51]: hmm... I think I mean to say "root directory", after reflection.

Comment (bgogul) [reply src/tools/tests/recipe2plan-test.ts]: Makes sense. TIL about android manifest conventions! :)

**Comment by alxrsngrtn** [comment]
Will merge after TAP is fixed and we can pass the presubmit.

**Pull request merged**